src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -646,7 +646,12 @@ public function dumpFile($filename, $content)
             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
         }
 
-        @chmod($tmpFile, 0666 & ~umask());
```

Comment:
I like the ternary op because it's not too long ... but I'd check a positive condition because negative conditions are less readable:

```php
@chmod($tmpFile, file_exists($filename) ? fileperms($filename) : 0666 & ~umask());
```